api/host.py

```diff
@@ -188,25 +185,23 @@ def _delete_host_list(host_id_list):
     ):
         query = _get_host_list_by_id_list_from_db(host_id_list)
 
-        if not query.count():
-            flask.abort(status.HTTP_404_NOT_FOUND)
-
         deletion_count = 0
-        for host_id, deleted in delete_hosts(
-            query, current_app.event_producer, inventory_config().host_delete_chunk_size
-        ):
-            if deleted:
-                log_host_delete_succeeded(logger, host_id, get_control_rule())
-                tracker_message = "deleted host"
-                deletion_count += 1
-            else:
-                log_host_delete_failed(logger, host_id, get_control_rule())
-                tracker_message = "not deleted host"
-
-            with PayloadTrackerProcessingContext(
-                payload_tracker, processing_status_message=tracker_message
-            ) as payload_tracker_processing_ctx:
-                payload_tracker_processing_ctx.inventory_id = host_id
+        if query:
```

Comment:
I don't think this works as expected; afaict the most performant way to verify that 1 or more records are returned is:
```suggestion
        if query.first()
```
Correct me if I'm wrong though @RonnyPfannschmidt

Reply:
@RonnyPfannschmidt I was taking a deeper look into this and saw that the query is a SQLAlchemy BaseQuery object, so even if no hosts have been found, query is True and the code goes into the for loop to try to delete the hosts. According to the good people of the internet Asa's suggestion is the best one to check for results.

Reply:
the point is - if we run the deletion, and there where no results , its jsut a empty loop

if there are results, we have to run the loop anyway, and thus have a extra count query

additionally count queries on postgresql can be expensive (for correctness constraints)

so why not jsut run the deletion function regardess of the count

Reply:
Ok, let me see if were on the same page. The actual deletion occurs on line 190, so if there are no results from `_get_host_list_by_id_list_from_db()`,  `deleted_count` stays as 0 and the whole deletion part is not executed. That way the deletion doesn't happen on a empty list. That was the previous behavior and what I tried to maintain without having to send a 404 to the user.
So the way I see it the deletion loop only happens if there are results from the db and I'm having a hard time understanding your point because of that.

Reply:
Ohhhhh I see what you mean now @RonnyPfannschmidt. Yes, since `delete_hosts()` will return no results in that scenario, we shouldn't need to check `query.count()` or `query.first()`. I'm in agreement that the conditional can just be removed.

@FabriciaDinizRH Since we already do `while select_query.count()` in [delete_hosts on L20](https://github.com/FabriciaDinizRH/insights-host-inventory/blob/ESSNTL-2359/lib/host_delete.py#L20), checking to see if any records are returned right before that is redundant. If no hosts are returned by that query, there are no records to iterate through, and the whole loop is effectively skipped.

Reply:
Oh, I see. I guess i was not looking at the right place.
Fixed it.